sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -360,6 +360,7 @@ trait CheckAnalysis extends PredicateHelper {
 
           case InsertIntoTable(t, _, _, _, _)
```

Comment:
This probably dates back to when we called these temp tables and when we didn't have persistent data source tables.  Given that I think we probably have to support this for compatibility.  I'm not sure why this isn't a whitelist though?